### PR TITLE
[bitnami/postgresql] Add extended configuration support for Read replica

### LINKED ISSUE #9472: [bitnami/postgresql] How to pass extendedConfiguration to replication ?
### Name and Version

bitnami/postgresql 11.0.2

### What steps will reproduce the bug?

1. Add Chart as a dependency :
```yaml
dependencies:
  - name: postgresql
    version: 11.0.2
    repository: ${PROXY_REPO_URL}
```
2. Use `architecture: replication` and `primary.extendedConfiguration`
```yaml
architecture: replication

primary:
  extendedConfiguration: |
    listen_addresses='0.0.0.0'
    port=5432
    max_connections=400
    shared_buffers='400MB'
    ...
```

### What is the expected behavior?

_No response_

### What do you see instead?

Note that those settings are only applied on primary and that no replication.extendedConfiguration is available  

Is it a bug or this feature isn't implemented yet ?

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ajammali**
**Description of the change**
Allow the support for extended configuration to the read replication.

**Benefits**
Add the ability to customize the PostgreSQL configuration file on the read replica.

**Possible drawbacks**
This still can't ensure that main configuration of the primary and read replicas is the same, which makes it impossible for the read to replace primary.

**Applicable issues**

  - fixes #9472

**Additional information**

**Checklist**

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ ] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [ ] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [ ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/postgresql/Chart.yaml [hunk 1]
```diff
@@ -26,4 +26,4 @@ name: postgresql
 sources:
   - https://github.com/bitnami/bitnami-docker-postgresql
   - https://www.postgresql.org/
-version: 11.1.22
+version: 11.1.23
```
Comment (miguelaeh) [inline bitnami/postgresql/Chart.yaml:29]:
```suggestion
version: 11.2.0
```

File: bitnami/postgresql/templates/_helpers.tpl [hunk 2]
```diff
@@ -33,7 +33,7 @@ Create the default FQDN for PostgreSQL read-only replicas headless service
 We truncate at 63 chars because of the DNS naming spec.
 */}}
 {{- define "postgresql.readReplica.svc.headless" -}}
-{{- printf "%s-hl" (include "postgresql.readReplica.fullname" .) | trunc 63 | trimSuffix "-" }}
+{{- printf "%s-hl" (include "postgresql.readReplicas.fullname" .) | trunc 63 | trimSuffix "-" }}
```
Comment (miguelaeh) [inline bitnami/postgresql/templates/_helpers.tpl:36]:
```suggestion
{{- printf "%s-hl" (include "postgresql.readReplica.fullname" .) | trunc 63 | trimSuffix "-" }}
```

Comment (miguelaeh) [reply bitnami/postgresql/templates/_helpers.tpl:36]: Seems that it was correct before the change, the function is called `readReplica`

File: bitnami/postgresql/templates/_helpers.tpl [hunk 3]
```diff
@@ -172,6 +172,27 @@ Return true if a configmap object should be created for PostgreSQL primary with
...
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return true if a configmap object should be created for PostgreSQL read with the extended configuration
+*/}}
+{{- define "postgresql.readReplica.createExtendedConfigmap" -}}
+{{- if and .Values.readReplicas.extendedConfiguration (not .Values.readReplicas.existingExtendedConfigmap) }}
+    {{- true -}}
+{{- else -}}
```
Comment (miguelaeh) [inline bitnami/postgresql/templates/_helpers.tpl:192]:
```suggestion
```

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Thanks for the PR!
I added some comments.
Also, I think it is confusing to use both `readReplica` and `readReplicas`, I suggest to use always the same to avoid confusion

**Comment by ajammali** [comment]
Thanks for the feedback.
I resolved the comments.

**Review by miguelaeh — COMMENTED** [review]
Please resolve the conflicts by rebasing master, I added the `verify` label so the actions are run.

**Comment by ajammali** [comment]
I think after rebasing master it all got messed up
if there is a way to correct this from your end or you can ignore this PR and you add the missing config your selves

**Comment by miguelaeh** [comment]
You can move between the local git states using `git reflog` to reset the branch to the commit before the rebase. 

After that simply checking out to master, making a pull to update master, and then checking out to your branch and rebasing should not cause problems unless your fork has other changes.

**Comment by ajammali** [comment]
> You can move between the local git states using `git reflog` to reset the branch to the commit before the rebase.
> 
> After that simply checking out to master, making a pull to update master, and then checking out to your branch and rebasing should not cause problems unless your fork has other changes.

the remote fork already has other changes from rebase. I'm not sure if it's a good idea to just delete the PR and start the whole thing all over

**Comment by rafariossaa** [comment]
I guess yes, that would be a better approach. 
Refer to this PR when you create the new one.

**Comment by ajammali** [comment]
I created a new PR #10165 to replace this one.

**Pull request closed without merging**